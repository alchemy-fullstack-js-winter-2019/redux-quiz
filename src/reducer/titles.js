import { UPDATE_TITLE, UPDATE_SUBTITLE } from '../actions/titles';

const initialState = {
  title: '',
  subtitle: ''
};

export default function reducer(state = initialState, action) {
  switch(action.type) {
    case UPDATE_TITLE: 
      return {
        ...state,
        title: action.payload
      };
    case UPDATE_SUBTITLE:
      return {
        ...state,
        subtitle: action.payload
      };
    default:
      return state;
  }
}
