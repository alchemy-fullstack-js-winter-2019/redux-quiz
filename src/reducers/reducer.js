import { UPDATE_TITLE, UPDATE_SUBTITLE } from '../actions/action';

const initialState = {
    title: '',
    subtitle: ''
};

export default function reducer(state =  initialState, action) {
    switch(action.type) {
        case UPDATE_TITLE:
            return {...state, title: action.payload }
        case UPDATE_SUBTITLE: 
            return{...state, subtitle: action.payload }
        default:
            return state; 
    }
};

