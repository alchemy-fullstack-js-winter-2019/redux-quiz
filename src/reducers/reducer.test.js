import reducer from './index';

describe('reducer tests', () => {
  it('tests our reducer', () => {
    const reducer1 = reducer('hello');

    expect(action1).toEqual({
      type: UPDATE_TITLE,
      payload: 'hello'
    });
  });
});
