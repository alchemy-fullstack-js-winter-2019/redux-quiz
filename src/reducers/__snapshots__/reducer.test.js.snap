// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer test reduer test 1`] = `
Object {
  "subtitle": "",
  "title": "title1",
}
`;
